packages/slice-machine/src/hooks/useServerState.tsx

```diff
@@ -2,20 +2,23 @@ import { useEffect } from "react";
 
 import ServerState from "@lib/models/server/ServerState";
 import useSliceMachineActions from "@src/modules/useSliceMachineActions";
+import useSwr from "swr";
+import axios, { AxiosResponse } from "axios";
 
-const useServerState = (serverState: ServerState) => {
-  const { getState } = useSliceMachineActions();
+const fetcher = (url: string): Promise<ServerState> =>
+  axios.get(url).then((res: AxiosResponse<ServerState>) => res.data);
+
+const useServerState = () => {
+  const { refreshState } = useSliceMachineActions();
+  const { data: serverState } = useSwr<ServerState>("/api/state", fetcher);
 
   useEffect(() => {
-    getState(serverState);
-  }, [
-    serverState.env,
-    serverState.warnings,
-    serverState.configErrors,
-    serverState.customTypes,
-    serverState.remoteCustomTypes,
-    serverState.libraries,
-  ]);
+    if (!serverState) {
+      return;
+    }
+
+    refreshState(serverState);
```

Comment:
what's refresh state do?

Reply:
send the data coming from the /state into the redux store